src/backend/distributed/utils/node_metadata.c

```diff
@@ -1013,7 +1007,7 @@ InsertNodeRow(int nodeid, char *nodeName, int32 nodePort, uint32 groupId, char *
 	values[Anum_pg_dist_node_noderole - 1] = ObjectIdGetDatum(nodeRole);
 
 	/* open shard relation and insert new tuple */
-	pgDistNode = heap_open(DistNodeRelationId(), AccessExclusiveLock);
+	pgDistNode = heap_open(DistNodeRelationId(), ShareRowExclusiveLock);
```

Comment:
Not very consequential, but we could probably just use `RowExclusiveLock` here since there's nothing in this function that specifically requires a `ShareRowExclusiveLock` (the caller already has one anyway).